### DISCUSSION TITLE
Feature: loading knowledge and problem dynamically

### DISCUSSION THREAD

**Discussion by robodrome** [PlanSys2/ros2_planning_system#270] [Ideas]
I would like to be able to load knowledge and problems more dynamically. I read your (@fmrico) comment about this [here](https://github.com/PlanSys2/ros2_planning_system/blob/f7ef8402b2dbc9610144c65c5a8fc66ff5e6181c/plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpertNode.cpp#L222-L223).

I think this is a must-have feature for robots that operate in more dynamic environments with - constantly - changing world states. I am right or am I missing something? Is this already possible right now?

**Comment by fmrico** [comment]
I think this could be very interesting. Take into account that loading a new domain needs to clean the problem expert content. Maybe we should do this via changing the lifecycle state of both modules active->inactive->unconfigured->inactive->active

**Reply by robodrome** [reply]
Yes, probably something like that. And some kind of trigger mechanism. 
- Perhaps domain and problem topics that receives pddl files. When a new pddl file is pushed, lifecycle is changed accordingly.
- When this happens the solver will generate a new plan automatically.

Perhaps it is better not to load the pddl's at launch at all and use topics in stead. Or am I overlooking something?
